org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JavaLanguageServerPlugin.java

```diff
@@ -142,6 +143,9 @@ public void start(BundleContext bundleContext) throws BackingStoreException {
 		preferenceManager.addPreferencesChangeListener(preferencesChangeListener);
 		logInfo(getClass() + " is started");
 		configureProxy();
+
+		// Set the ID to use for preference lookups
+		JavaManipulation.setJavaUIPluginId(PLUGIN_ID);
```

Comment:
How about you rename that method upstream to something like setPreferenceStoreId or setPreferenceStorePluginId?

Reply:
This change should be in for Photon M6, https://git.eclipse.org/r/118209/ . New methods will be getPreferenceNodeId/setPreferenceNodeId . The setter returns true if the value was successfully set and false otherwise.